### PR TITLE
Bug/sc 99081 escape dots using json path

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

**Review by MayeulRousselet — CHANGES_REQUESTED** [review]

File: code-env/python/spec/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+jsonpath_ng
```
[the remark this answers is not in the document]
Comment (alexbourret) [reply code-env/python/spec/requirements.txt]: 👍 done in [2200987](https://github.com/dataiku/dss-plugin-api-connect/pull/25/commits/22009879c67f325b7ae6ab0958d577c18ea18d61)

**Review by liamlynch-data — COMMENTED** [review]
The code looks fine to me - I will do some testing now.

I have one question/thought though - is it maybe worth taking the same approach for where the counting_key and extraction_key are split and used to call get_value_from_path()? get_value_from_path() in dku_utils.js is similar to the function you replaced - so these keys will probably suffer the same potential problem. (Maybe we don't want to to risk changing these things now though, at least not for this ticket)

**Comment by alexbourret** [comment]
Yes, the next version will use the same approach for all the "path style" parameters. But there are ramifications and integration tests to write for the "key to array" box. This version is a quick fix for a pagination issue.

**Review by liamlynch-data — APPROVED** [review]
Test details below - all working fine.

Tested against APIS with the two styles of next page urls as follows for both datasets and recipes, using a plugin built from the branch.

Paths with keys with dots:
```
    "Paging": {
        "@odata.nextLink": "https://bigsta.sh/api/dku-qa/pagination/sc-99081?page=2"
    }
```

in this example, all 3 of the following work in the plugin UI:
`Paging.['@odata.nextLink']`
`Paging."@odata.nextLink"`
`Paging.'@odata.nextLink'`

Regression - existing paths with dots in the path (not in any of the keys)
```
    "Paging": {
        "Next": "https://bigsta.sh/api/dku-qa/pagination/sc-76640?page=3"
    }
```

**Review by MayeulRousselet — APPROVED** [review]

**Pull request merged**